indexers/src/services/relayer.service.ts

```diff
@@ -0,0 +1,142 @@
+import { EthereumResult, EthereumTransaction } from "@subql/types-ethereum";
+import { SupportedChain } from "../types/enums";
+import { Relayer, Transfer } from "../types/models";
+import { RelayerChainStatsService } from "./relayerChainStats.service";
+import { getNativeCurrencyPrice } from "../utils/price.helpers";
+import {
+  HandlePostRequestsTransaction,
+  HandlePostResponsesTransaction,
+} from "../types/abi-interfaces/HandlerV1Abi";
+import { HyperBridgeService } from "./hyperbridge.service";
+
+export class RelayerService {
+  /**
+   * Find a relayer by its id or create a new one if it doesn't exist
+   */
+  static async findOrCreate(
+    relayer_id: string,
+    chain: SupportedChain,
+  ): Promise<Relayer> {
+    let relayer = await Relayer.get(relayer_id);
```

Comment:
This is not working as intended, if a relayer already exists, we need to check if the chain passed is already part of it's network list and update it here, then increase the number of unique relayers for that chain.

Without this the relayer will never reflect on the other chains it's relaying to, since in it's current state, this function can only be executed once.

Reply:
Done